### PR TITLE
Add form

### PR DESCRIPTION
**Pull request by MrBroadie**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
         "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx", ".ts", ".tsx"] }],
         "semi": [2, "always"],
         "indent": ["error", 2],
-        "object-curly-spacing": ["error", "always"]
+        "object-curly-spacing": ["error", "always"],
+        "@typescript-eslint/no-explicit-any": "off"
```
Comment (nicallennn) [inline .eslintrc.json]: i have removed this now as removed all the any's, can you remove and repush pls

File: src/components/antdComponents/antdform/AntdForm.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  const displayForm = props.data.map((form) => (
+    <Form.Item name={form.name} key={form.name} rules={form.rules}> 
+      {form.content}
+    </Form.Item>
+  ));
+  
+
+  return (
+    <div className='genericContainer'>
+      <Form
```
Comment (nicallennn) [inline src/components/antdComponents/antdform/AntdForm.tsx:38]: the form should be inside the inner container else it wont get the padding. unless there is a specific reason for this?

File: src/components/antdComponents/antdform/AutoCompleteForm.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { AutoComplete, Form } from 'antd';
+import { useState } from 'react';
+
+const mock = [{ value: 'AMD' }, { value: 'AAPL' }, { value: 'NIO' }, { value: 'SOFI' }, { value: 'DAME' }, { value: 'DAMV' }];
+
+const AutoCompleteForm: React.FC = () => {
```
Comment (nicallennn) [inline src/components/antdComponents/antdform/AutoCompleteForm.tsx]: does this need a type? should we be doing this everywhere?

File: src/components/antdComponents/antdform/AutoCompleteForm.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  const [options, setOptions] = useState<{ value: string }[]>([]);
+  const onSearch = (searchText: string) => {
+    !searchText ? setOptions([]) : [mock.forEach(el => (el.value.includes(searchText.toUpperCase()) && !options.includes(el)) && setOptions([el]) )];
+  };
+  const onSelect = (data: string) => {
+    console.log('data', data);
+  };
+
+  return (
+    <Form.Item  name='symbol' rules={[{ required: true, message: 'Please input symbol!' },{ max: 4, message: 'Symbol must not exceed 4 characters' }]} >
```
Comment (nicallennn) [inline src/components/antdComponents/antdform/AutoCompleteForm.tsx]: can you extract the rules out to a variable pls

File: src/components/default/styles/DefaultNavbar.scss [hunk 5]
```diff
@@ -1,6 +1,6 @@
-@use '../../../sassVariables/colors.scss' as colors;
-@use '../../../sassVariables/spacing.scss' as spacing;
-@use '../../../sassVariables//fonts.scss' as fonts;
+@use 'sassVariables/colors.scss' as colors;
```
Comment (nicallennn) [inline src/components/default/styles/DefaultNavbar.scss:1]: nice :)

File: src/components/antdComponents/antdform/AntdForm.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  }[];
+}
+
+const AntdForm = (props: AntdFormProps) => {
+
+  const { user } = useAuth0();
+
+  // Inside this function we call api service. Need to check if the SYMBL exists
+  const onSubmit = (values) => {
+    console.log('Success:', values, user?.sub);
```
Comment (nicallennn) [inline src/components/antdComponents/antdform/AntdForm.tsx:22]: perfect thank you, will add this service in tomorrow, can pair program it if you like

File: src/components/authenticatedComponents/MobileNavbarAuth.tsx [hunk 7]
```diff
@@ -35,7 +35,7 @@ const MobileNavbarAuth = () => {
         Profile
         </Link>
 
-        <Link to="/projects" className="mobileLink subLink" onClick={handleMenuClick}>
+        <Link to="/addInvestments" className="mobileLink subLink" onClick={handleMenuClick}>
```
Comment (nicallennn) [inline src/components/authenticatedComponents/MobileNavbarAuth.tsx:38]: can you remind me about sorting the navigation out when we have all the pages worked out and running pls :)

File: src/pages/authenticatedPages/AddInvestment/AddInvestmentForm.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
+// Imports
```
Comment (nicallennn) [inline src/pages/authenticatedPages/AddInvestment/AddInvestmentForm.tsx:1]: this all looks cool

File: src/pages/authenticatedPages/AddInvestment/styles/AddInvestment.scss [hunk 9]
```diff
@@ -0,0 +1,26 @@
+@use 'sassVariables/colors.scss' as colors;
+@use 'sassVariables/spacing.scss' as spacing;
+@use 'sassVariables/fonts.scss' as fonts;
+
+.genericContainer {
+  .genericInnerContainer{
```
Comment (nicallennn) [inline src/pages/authenticatedPages/AddInvestment/styles/AddInvestment.scss:5]: idk if you need to target these but we can review this later.

**Review by nicallennn — CHANGES_REQUESTED** [review]
this looks nice overall, nice work :), i have made a few requests for changes, lemme know if you need more info

File: src/components/antdComponents/antdform/AntdForm.tsx (hunk 2 above)
Comment (MrBroadie) [reply src/components/antdComponents/antdform/AntdForm.tsx:38]: Yep form is the outer component. All of the Form Items are inside the Inner container. I had to do it this way to get the button outside of the Inner container whilst also keeping it semi grouped.

File: src/components/antdComponents/antdform/AutoCompleteForm.tsx (hunk 3 above)
Comment (MrBroadie) [reply src/components/antdComponents/antdform/AutoCompleteForm.tsx]: I believe this is the way of things --> This is the recommended way on ant d so prehaps we look into changing all out functions to this tomorrow

**Review by nicallennn — APPROVED** [review]

**Pull request merged**